File: Sources/UIInspector/UIInspector.swift
import UIKit

/// UIInspector – iOS アプリ向けUIデバッグツール。
///
/// Quick start:
/// ```swift
/// // AppDelegate or SwiftUI App.init:
/// #if DEBUG
/// UIInspector.shared.install()
/// #endif
/// ```
@MainActor
public final class UIInspector {
    public static let shared = UIInspector()
    private var manager: InspectorManager?

    private init() {}

    /// インスペクタをインストールする。追加プラグインを渡すことで拡張可能。
    /// Built-in: View Hierarchy / View Attributes / Measure
    public func install(additionalPlugins: [any InspectorPlugin] = []) {
        let defaults: [any InspectorPlugin] = [
            ViewHierarchyPlugin(),
            ViewAttributesPlugin(),
            MeasurePlugin()
        ]
        let mgr = InspectorManager(plugins: defaults + additionalPlugins)
        mgr.setup()
        self.manager = mgr
    }

    /// プログラムからプラグインメニューを開く。
    public func show() {
        manager?.showPluginMenu()
    }

    /// インスペクタを閉じる。
    public func hide() {
        manager?.deactivate()
    }
}

File: android/uiinspector/src/main/kotlin/com/uiinspector/UIInspector.kt
package com.uiinspector

import android.app.Application
import com.uiinspector.core.InspectorManager
import com.uiinspector.core.InspectorPlugin
import com.uiinspector.plugins.MeasurePlugin
import com.uiinspector.plugins.ViewAttributesPlugin
import com.uiinspector.plugins.ViewHierarchyPlugin

object UIInspector {

    private var manager: InspectorManager? = null

    fun install(app: Application, additionalPlugins: List<InspectorPlugin> = emptyList()) {
        if (manager != null) return
        val defaultPlugins = listOf(
            ViewHierarchyPlugin(),
            ViewAttributesPlugin(),
            MeasurePlugin()
        )
        val m = InspectorManager(defaultPlugins + additionalPlugins)
        app.registerActivityLifecycleCallbacks(m)
        manager = m
    }

    fun show() {
        manager?.showPluginMenu()
    }
}
